skbio/stats/distance/_regression.py

```diff
@@ -0,0 +1,170 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2013--, scikit-bio development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+from __future__ import absolute_import, division, print_function
+
+import six
+import numpy as np
+from skbio.util._decorator import experimental
+from skbio.util._misc import check_random_state
+from skbio.stats.distance._mantel import _order_dms
+
+from skbio.stats.distance import DistanceMatrix
+
+@experimental(as_of="0.4.0")
+def linregress(y, *args, **kwargs):
+    """
+    This module performs a multiple linear regression on distance
+    matrices.  This is done by extracting the upper (or lower)
+    triangular portions of the symmetric distance matrices,
+    and collapsing them to compute a multiple linear regression
+    on the distances.  The signficances are computed by
+    permuting the residuals.
+
+    Parameters
+    ----------
+    y : DistanceMatrix, array_like, or filepath
```

Comment:
@mortonjt I think you should remove filepath support and just accept `DistanceMatrix` and `array_like` since they'll all have to be loaded into memory at once.